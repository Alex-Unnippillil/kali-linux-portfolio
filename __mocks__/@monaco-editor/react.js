import React from 'react';
export default function MonacoMock({ value = '', onChange }) {
  return React.createElement('textarea', {
    value,
    onChange: e => onChange && onChange(e.target.value),
  });
}
