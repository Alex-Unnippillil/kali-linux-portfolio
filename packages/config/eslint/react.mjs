import globals from 'globals';
import { baseConfig, compat } from './base.mjs';

const reactConfig = [
  ...baseConfig,
  {
    languageOptions: {
      globals: globals.browser,
    },
    rules: {
      'no-top-level-window/no-top-level-window-or-document': 'off',
      'no-undef': 'off',
      'no-unused-vars': 'off',
      'react/react-in-jsx-scope': 'off',
    },
  },
  ...compat.config({
    extends: [
      'eslint:recommended',
      'plugin:@typescript-eslint/recommended',
      'plugin:react/recommended',
      'plugin:react-hooks/recommended',
    ],
    settings: {
      react: {
        version: 'detect',
      },
    },
  }),
  {
    rules: {
      '@typescript-eslint/no-unused-vars': ['error', { argsIgnorePattern: '^_' }],
      '@typescript-eslint/no-unused-expressions': ['error', { allowShortCircuit: true }],
      'react/react-in-jsx-scope': 'off',
    },
  },
];

export default reactConfig;
