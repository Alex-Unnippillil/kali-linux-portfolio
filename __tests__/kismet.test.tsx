import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import KismetApp from '../components/apps/kismet';

describe('KismetApp', () => {
  it('steps through sample capture frames', async () => {
    const user = userEvent.setup();
    render(<KismetApp />);
    await user.click(screen.getByRole('button', { name: /load sample/i }));
    await user.click(screen.getByRole('button', { name: /step/i }));
    expect(screen.getAllByText('CoffeeShopWiFi')[0]).toBeInTheDocument();
  });

  it('invokes onNetworkDiscovered with network info', async () => {
    const user = userEvent.setup();
    const onNetworkDiscovered = jest.fn();
    render(<KismetApp onNetworkDiscovered={onNetworkDiscovered} />);
    await user.click(screen.getByRole('button', { name: /load sample/i }));
    await user.click(screen.getByRole('button', { name: /step/i }));
    expect(onNetworkDiscovered).toHaveBeenCalledWith(
      expect.objectContaining({
        ssid: 'CoffeeShopWiFi',
        bssid: '66:77:88:00:00:01',
        discoveredAt: expect.any(Number),
      })
    );
  });
});
