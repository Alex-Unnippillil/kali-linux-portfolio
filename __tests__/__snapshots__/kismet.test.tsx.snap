// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`KismetPage layout renders simulated sections with context banner 1`] = `
<div>
  <div
    class="space-y-10 lg:space-y-12"
  >
    <aside
      aria-label="Simulation notice"
      class="rounded-xl border border-amber-500/40 bg-amber-900/20 p-5 text-sm text-amber-100/90 shadow"
      role="note"
    >
      <p
        class="text-base font-semibold uppercase tracking-wide text-amber-100"
      >
        Simulation notice
      </p>
      <p
        class="mt-2"
      >
        All telemetry in this workspace is replayed from bundled fixture data (
        <code
          class="font-mono"
        >
          sampleCapture.json
        </code>
         and
         
        <code
          class="font-mono"
        >
          sampleClients.json
        </code>
        ) so the demo can run entirely offline.
      </p>
      <p
        class="mt-2"
      >
        Safe Mode keeps radio controls, uploads, and outbound probes disabled; toggling Lab Mode only unlocks additional filters on the simulated dataset.
      </p>
    </aside>
    <section
      aria-labelledby="kismet-monitor-heading"
      class="rounded-2xl border border-slate-700/70 bg-slate-950/70 p-8 shadow-xl"
    >
      <h1
        class="mb-6 text-3xl font-bold tracking-tight text-sky-50 sm:text-4xl"
        id="kismet-monitor-heading"
      >
        Kismet Live Monitor (Simulated)
      </h1>
      <div
        class="space-y-6 p-4 text-[color:var(--kali-text)] lg:space-y-8"
      >
        <div
          class="rounded-xl border border-kali-border/70 bg-[color:var(--kali-surface)] p-4 text-sm shadow-kali-panel"
          role="status"
        >
          <div
            class="flex flex-col gap-3 lg:flex-row lg:items-center lg:justify-between"
          >
            <div
              class="space-y-1"
            >
              <p
                class="text-base font-semibold uppercase tracking-wide text-kali-accent"
              >
                Lab Mode 
                off
              </p>
              <p
                class="text-xs text-kali-muted"
              >
                Enable Lab Mode to explore the full simulator. Offline summary remains available.
              </p>
            </div>
            <div
              class="flex flex-wrap gap-2"
            >
              <button
                class="rounded-lg bg-[color:var(--color-accent)] px-3 py-1.5 text-xs font-semibold uppercase tracking-wide text-[color:var(--color-inverse)] shadow-sm transition hover:bg-[color:color-mix(in_srgb,var(--color-accent)_88%,transparent)] focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-kali-focus"
                type="button"
              >
                Enable Lab Mode
              </button>
              <button
                class="rounded-lg border border-kali-border/70 px-3 py-1.5 text-xs font-semibold uppercase tracking-wide text-kali-accent transition hover:bg-[color:color-mix(in_srgb,var(--color-accent)_12%,transparent)] focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-kali-focus"
                type="button"
              >
                Reload fixtures
              </button>
            </div>
          </div>
        </div>
        <div
          class="rounded-xl border border-kali-border/70 bg-[color:var(--kali-surface)] p-4 text-xs shadow-kali-panel"
        >
          <p
            class="text-sm font-semibold uppercase tracking-wide text-kali-accent"
          >
            Offline summary
          </p>
          <ul
            class="mt-3 list-disc space-y-1 pl-5 text-[color:color-mix(in_srgb,var(--kali-text)_82%,transparent)]"
          >
            <li>
              Access points discovered: 
              4
            </li>
            <li>
              Total beacon frames observed: 
              6
            </li>
            <li>
              Busiest channel:
               
              1
            </li>
          </ul>
        </div>
      </div>
    </section>
    <section
      aria-labelledby="deauth-walkthrough-heading"
      class="rounded-2xl border border-slate-700/70 bg-slate-950/70 p-8 shadow-xl"
    >
      <h2
        class="mb-4 text-2xl font-semibold tracking-tight text-sky-50"
        id="deauth-walkthrough-heading"
      >
        Deauthentication Walkthrough
      </h2>
      <div
        class="space-y-4 text-sm"
      >
        <table
          class="w-full text-sm"
        >
          <thead>
            <tr
              class="text-left"
            >
              <th
                class="pr-2"
              >
                Seq
              </th>
              <th
                class="pr-2"
              >
                Source
              </th>
              <th
                class="pr-2"
              >
                Destination
              </th>
              <th>
                Type
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              class="odd:bg-gray-800 "
            >
              <td
                class="pr-2"
              >
                1
              </td>
              <td
                class="pr-2"
              >
                66:77:88:00:00:01
              </td>
              <td
                class="pr-2"
              >
                11:22:33:44:55:66
              </td>
              <td>
                Data
              </td>
            </tr>
            <tr
              class="odd:bg-gray-800 "
            >
              <td
                class="pr-2"
              >
                2
              </td>
              <td
                class="pr-2"
              >
                66:77:88:00:00:01
              </td>
              <td
                class="pr-2"
              >
                11:22:33:44:55:66
              </td>
              <td>
                Data
              </td>
            </tr>
            <tr
              class="odd:bg-gray-800 bg-red-900 text-red-100"
            >
              <td
                class="pr-2"
              >
                3
              </td>
              <td
                class="pr-2"
              >
                66:77:88:00:00:01
              </td>
              <td
                class="pr-2"
              >
                FF:FF:FF:FF:FF:FF
              </td>
              <td>
                Deauth
              </td>
            </tr>
            <tr
              class="odd:bg-gray-800 bg-red-900 text-red-100"
            >
              <td
                class="pr-2"
              >
                4
              </td>
              <td
                class="pr-2"
              >
                66:77:88:00:00:01
              </td>
              <td
                class="pr-2"
              >
                FF:FF:FF:FF:FF:FF
              </td>
              <td>
                Deauth
              </td>
            </tr>
            <tr
              class="odd:bg-gray-800 "
            >
              <td
                class="pr-2"
              >
                5
              </td>
              <td
                class="pr-2"
              >
                66:77:88:00:00:01
              </td>
              <td
                class="pr-2"
              >
                11:22:33:44:55:66
              </td>
              <td>
                Data
              </td>
            </tr>
          </tbody>
        </table>
        <p
          class="mt-4"
        >
          Frames 3 and 4 show back-to-back deauthentication messages, a pattern often used to forcibly disconnect clients from an access point.
        </p>
        <p
          class="mt-2"
        >
          For defensive guidance, review the
           
          <a
            class="text-blue-400 underline"
            href="/docs/latest/deauth-mitigation"
          >
            mitigation notes (latest docs)
          </a>
          .
        </p>
      </div>
    </section>
  </div>
</div>
`;
